### PR TITLE
Convert "Customise media actions" to Compose

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

This change converts the "Customise media actions" page to Compose. 

## Testing Instructions

- Play an episode
- Tap on the Profile tab
- Tap on the settings cog
- Tap on General
- Tap on Media notification controls
- ✅ Pull down the notification shade and check the actions match (you might only be able to see two)
- On the "Media notification controls" page drag to rearrange the actions
- ✅ Pull down the notification shade and check the actions match
- Tap on the "Show custom media actions" to disable the actions
- ✅ Verify you can drag the actions and check the notification actions are gone 

## Screenshots

| Before | After |
| --- | --- |
| Screenshot_20240906_214254 | Screenshot_20240906_220602 | 
| Screenshot_20240906_214310 | Screenshot_20240906_220616 |

https://github.com/user-attachments/assets/c77cd70d-0cee-4f37-a13f-f27fef322ccf

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          3 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>MediaActionsViewModel</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>State</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>MenuAction</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/rearrange/MenuActionRearrange.kt [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                        color = titleTextColor,
+                        modifier = Modifier.padding(24.dp),
+                    )
+                }
+            }
+        }
+
+        menuActions.forEachIndexed { index, action ->
+            if (index == chooseCount) {
+                stickyHeader {
```
Comment (ashiagr) [inline modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/rearrange/MenuActionRearrange.kt:112]: Nice!

**Review by ashiagr — APPROVED** [review]
Works great!

I also tested if changes to the menu are properly retained on upgrading the app. 

Just left a few nitpicks.

File: modules/services/localization/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1217,7 +1219,7 @@
     <string name="settings_media_actions_prioritize_subtitle">Your top actions will be available.</string>
     <string name="settings_media_actions_customise">Customize media actions</string>
     <string name="settings_media_actions_show_title">Show custom media actions</string>
-    <string name="settings_media_actions_show_subtitle">Show custom media actions in Android 13 media notifications and Android Auto.</string>
+    <string name="settings_media_actions_show_subtitle">Show custom media actions in Android 13+ media notifications and Android Auto.</string>
```
[the remark this answers is not in the document]
Comment (geekygecko) [reply modules/services/localization/src/main/res/values/strings.xml:1222]: Thanks for the suggestions.

**Pull request merged**